include/miopen/miopen.h

```diff
@@ -2531,6 +2532,35 @@ MIOPEN_EXPORT miopenStatus_t miopenLayerNormForward(miopenHandle_t handle,
 // CLOSEOUT LAYERNORM DOXYGEN GROUP
 #endif
 
+#ifdef MIOPEN_BETA_API
+// Cat APIs
+/** @addtogroup cat
+ *
+ *  @{
+ */
+/*! @brief Execute a cat forward layer
+ *
+ * This API only implements the LAYERNORM_MODE_CHANNEL in LAYERNORM_ACCURATE path.
+ *
+ * @param handle         MIOpen handle (input)
+ * @param xDescs         Tensor descriptor of input tensor x (input)
+ * @param xs             Source data tensor x (input)
+ * @param yDesc          Tensor descriptor of output tensor y (input)
+ * @param y              Data tensor y (output)
+ * @param dim            Concatenation dimension (input)
+ * @return               miopenStatus_t
+ */
+MIOPEN_EXPORT miopenStatus_t miopenCatForward(miopenHandle_t handle,
+                                              const std::vector<miopenTensorDescriptor_t>& xDescs,
```

Comment:
The API cannot have C++ constructs.

Reply:
Moreover, it would be a better design to use the existing MIOpen TensorOp API with a new enum for the op itself.

Reply:
- Convert vector to pointer array and array size.
- The TensorOP is accepted only two input tensors, whereas the Cat requires a variable number of input tensors and a specified concatenation dimension. The formats of the two API differ.